climada_petals/hazard/test/test_tc_rainfield.py

```diff
@@ -97,55 +127,64 @@ def test_from_file_pass(self):
-        tc_haz = TCRain.from_tracks(tc_track, CENTR_TEST_BRB)
+        tc_haz = TCRain.from_tracks(tc_track, centroids=CENTR_TEST_BRB)
         tc_haz.remove_duplicates()
         tc_haz.check()
 
         self.assertEqual(tc_haz.haz_type, 'TR')
         self.assertEqual(tc_haz.tag.description, [])
-        self.assertEqual(tc_haz.tag.file_name, ['IBTrACS: 1951239N12334'])
+        self.assertEqual(tc_haz.tag.file_name, ['Name: 1951239N12334'])
         self.assertEqual(tc_haz.units, 'mm')
         self.assertEqual(tc_haz.centroids.size, 296)
         self.assertEqual(tc_haz.event_id.size, 1)
         self.assertEqual(tc_haz.event_id[0], 1)
         self.assertEqual(tc_haz.event_name, ['1951239N12334'])
         self.assertTrue(np.array_equal(tc_haz.frequency, np.array([1])))
         self.assertTrue(np.array_equal(tc_haz.orig, np.array([True])))
-        self.assertTrue(isinstance(tc_haz.intensity, sparse.csr.csr_matrix))
-        self.assertTrue(isinstance(tc_haz.fraction, sparse.csr.csr_matrix))
-        self.assertEqual(tc_haz.intensity.shape, (1, 296))
+        self.assertTrue(isinstance(tc_haz.fraction, sparse.csr_matrix))
         self.assertEqual(tc_haz.fraction.shape, (1, 296))
-
-        self.assertEqual(tc_haz.fraction.nonzero()[0].size, 0)
+        self.assertIsNone(tc_haz._get_fraction())
+        self.assertTrue(isinstance(tc_haz.intensity, sparse.csr_matrix))
+        self.assertEqual(tc_haz.intensity.shape, (1, 296))
         self.assertEqual(tc_haz.intensity.nonzero()[0].size, 0)
 
 class TestModel(unittest.TestCase):
     """Test modelling of rainfall"""
 
-    def test_rainfield_from_track_pass(self):
-        """Test _rainfield_from_track function. Compare to MATLAB reference."""
+    def test_compute_rain_pass(self):
+        """Test _compute_rain function. Compare to MATLAB reference."""
         tc_track = TCTracks.from_processed_ibtracs_csv(TEST_TRACK)
         tc_track.equal_timestep()
-        rainfall = rainfield_from_track(tc_track.data[0], CENTR_TEST_BRB)
+        track_ds = tc_track.data[0]
+        si_track = _track_to_si_with_q_and_shear(track_ds)
+        centroids = CENTR_TEST_BRB
+        ncentroids = centroids.size
+        rainfall = np.zeros((1, ncentroids))
+        rainrates, reachable_centr_idx = compute_rain(
+            si_track, centroids.coord, MODEL_RAIN["R-CLIPER"],
+        )
+        rainfall[0, reachable_centr_idx] = (
+            (rainrates * track_ds["time_step"].values[:, None]).sum(axis=0)
+        )
 
         rainfall = np.round(rainfall, decimals=9)
 
-        self.assertAlmostEqual(rainfall[0, 0], 66.801702386)
-        self.assertAlmostEqual(rainfall[0, 130], 43.290917792)
-        self.assertAlmostEqual(rainfall[0, 200], 76.315923838)
+        self.assertAlmostEqual(rainfall[0, 0], 65.114948501)
+        self.assertAlmostEqual(rainfall[0, 130], 39.584947656)
+        self.assertAlmostEqual(rainfall[0, 200], 73.792450959)
```

Comment:
This change in intensity values for the R-CLIPER model is also due to the bug fix described in the comments above.